TombEngine/Game/items.cpp

```diff
@@ -838,61 +838,56 @@ void DefaultItemHit(ItemInfo& target, ItemInfo& source, std::optional<GameVector
 	DoItemHit(&target, damage, isExplosive);
 }
 
-
-//TODO: Originally was using a ItemInfo, but has been simplified to use only vector3i. Is other .cpp more suitable to contain this function?
+// TODO: Originally took an ItemInfo argument, but has been simplified to use only Vector3i.
+// Is other .cpp more suitable to contain this function?
 Vector3i GetNearestSectorCenter(const Vector3i& pos)
 {
-	const unsigned int maskLower8Bits = 0xFFFFFE00;	// mask for the bits [0-8]
-	const unsigned int mask9Bit = 0x200;			// mask for the bit [9] (512)
+	constexpr auto BIT_MASK_LOWER_8 = 0xFFFFFE00; // 0-8
+	constexpr auto BIT_MASK_9		= 0x200;	  // 9
 
-	//Clears the 0-8 bits and set the bit 9 to put 512 at the end. (Placing the object in the sector center).
-	return Vector3i (
-					 pos.x & maskLower8Bits | mask9Bit,
-					 pos.y,
-					 pos.z & maskLower8Bits | mask9Bit
-					);
+	// Return collision block center.
+	// TODO: No bitwise operations.
+	return Vector3i(
+		pos.x & BIT_MASK_LOWER_8 | BIT_MASK_9,
+		pos.y,
+		pos.z & BIT_MASK_LOWER_8 | BIT_MASK_9);
```

Comment:
It can be done with maths, but I let the bitwise operation because although ugly, they were more optimal.

So I created this function to move the bitwise ugliness here instead of the object control function.

If decided to apply with maths in a readable way, I think this function could be discarded (is not used a lot, I think like 4 or 5 times along the engine).

Reply:
It's a fairly high-level function, so optimisation of that sort isn't too necessary.

Reply:
The function is also handy so I wouldn't discard it. :)